### PR TITLE
Updating registration. User can now enter additional optional information to his profile on registration.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
User has to provide some mandatory information to register. In addition to that, users can now provide further optional information for their profiles and they will be setup in user's profile upon successful signup.

### CODE REVIEW

**Comment by saadyousafarbi** [comment]
@zubair-arbi
please have a look at the update.

File: autoshop/core/forms.py [hunk 1]
```diff
@@ -14,11 +14,24 @@ class SignupForm(forms.Form):
...
-    last_name = forms.CharField(max_length=30)
-    email = forms.EmailField(max_length=100)
-    username = forms.CharField(max_length=30)
-    password = forms.CharField(widget=forms.PasswordInput, required=True)
+    GENDER_CHOICES = (
+        ('M', 'Male'),
+        ('F', 'Female'),
+    )
+
+    first_name = forms.CharField(help_text='Enter first name', max_length=30)
```
Comment (zubair-arbi) [inline autoshop/core/forms.py]: Why not make it a `User` model form then add additional fields for profile.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
One suggestion otherwise looks good.

File: autoshop/core/forms.py (hunk 1 above)
Comment (saadyousafarbi) [reply autoshop/core/forms.py]: you mean extending the form from UserCreationForm?

File: autoshop/core/forms.py [hunk 2]
```diff
@@ -14,11 +15,27 @@ class SignupForm(forms.Form):
...
-    first_name = forms.CharField(max_length=30)
-    last_name = forms.CharField(max_length=30)
-    email = forms.EmailField(max_length=100)
-    username = forms.CharField(max_length=30)
-    password = forms.CharField(widget=forms.PasswordInput, required=True)
+    GENDER_CHOICES = (
+        ('M', 'Male'),
+        ('F', 'Female'),
+    )
+    first_name = forms.CharField(help_text='Enter first name', max_length=30)
```
Comment (zubair-arbi) [inline autoshop/core/forms.py]: if we are using `User` model as model form then do we need to declare/overwrite the fields again here?

Comment (saadyousafarbi) [reply autoshop/core/forms.py]: If i wish to have my own help_text for the User fields, what should i do then?

Comment (zubair-arbi) [reply autoshop/core/forms.py]: Lets update the help text of model

File: autoshop/core/forms.py [hunk 3]
```diff
@@ -14,11 +15,32 @@ class SignupForm(forms.Form):
...
+    city = forms.CharField(help_text='Enter city', required=False)
+    country = forms.CharField(help_text='Enter country', required=False)
+
+    class Meta:
+        model = User
+        fields = ['first_name', 'last_name', 'username', 'email', 'password']
+        widgets = {
+            'password': forms.PasswordInput(),
+        }
+        help_texts = {
```
Comment (zubair-arbi) [inline autoshop/core/forms.py]: nice.

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍 
Please rebase and squash before merge. Also dont forget to add comment on PR about your squashing from last commit (with hash).

**Comment by saadyousafarbi** [comment]
Squashed from commit a67e03dd052fd6541949230cc9295ca3968cd400

**Pull request merged**